contrib/fuzz/archive_fuzz_test.go

```diff
@@ -0,0 +1,145 @@
+/*
+   Copyright The containerd Authors.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package fuzz
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"path"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+
+	imageArchive "github.com/containerd/containerd/v2/core/images/archive"
+	"github.com/containerd/containerd/v2/pkg/archive"
+	"github.com/containerd/containerd/v2/plugins/content/local"
+	"github.com/containerd/log"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// FuzzApply implements a fuzzer that applies
+// a fuzzed tar archive on a directory
+func FuzzApply(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		ctx := context.Background()
+
+		// Apply() is logging the below message, which is too noisy and not really useful
+		// if the input is random.
+		//
+		// level=warning msg="ignored xattr ... in archive" error="operation not supported"
+		log.G(ctx).Logger.SetLevel(log.PanicLevel)
+
+		f := fuzz.NewConsumer(data)
+		iters, err := f.GetInt()
+		if err != nil {
+			return
+		}
+		maxIters := 20
+		tmpDir, err := os.MkdirTemp("", "prefix-test")
```

Comment:
Suggestion: Use `t.TempDir()` here and all other places.

Reply:
Fixed